src/lem/interpreter.rs

```diff
@@ -97,13 +111,35 @@ impl Block {
                     let tgt_ptr = src_ptr.cast(*tag);
                     bindings.insert(tgt.clone(), tgt_ptr);
                 }
+                Op::EqTag(tgt, a, b) => {
+                    let a = bindings.get(a)?;
+                    let b = bindings.get(b)?;
+                    let c = if a.tag() == b.tag() {
+                        Ptr::Leaf(Tag::Expr(Num), F::ONE)
+                    } else {
+                        Ptr::Leaf(Tag::Expr(Num), F::ZERO)
+                    };
```

Comment:
Can you help me understand how these values will be used? It looks like you're creating Lurk `Num`s with values corresponding to `Boolean` (i.e. in circuit) values. While maybe this works as you're using it (not sure yet), it also seems to be a 'mixed metaphor'. The Lurk values `1` and `0` are not used to signal truth value. And real `AllocatedBit`s must be constrained as such.

Again, I'm not saying that anything is 'wrong' per se — just noting the possible bugs; while also stating that I think this probably *is* wrong from the perspective that some ad-hoc values that probably should never exist are being created; and I don't think that should happen in LEM. It's worrisome and at best will lead to confusion in the future.

Reply:
This is a good question. Since LEM does not yet have types (but it will eventually do!), the LEM programmer will have to keep track of which variables are actually Nums and which are Booleans. I eventually want to add a cheap `if` statement, that does not allocate a selector boolean, but simply uses the boolean it takes as input

Reply:
In this example, the return value of an `EqTag` operation should be a boolean. If you look at the circuit implementation, I'm having to transform the boolean to a num, because of the lack of types